### PR TITLE
DISPATCH-1977: Introduced atomics raw_closed_read and raw_closed_writ…

### PR DESCRIPTION
**Pull request by ganeshmurthy**
…e in http2 adaptor. Also made raw_closed_write and raw_closed_read as atomics in tcp_adaptor"

### CODE REVIEW

File: include/qpid/dispatch/atomic.h [hunk 1]
```diff
@@ -205,6 +205,9 @@ static inline void sys_atomic_destroy(sys_atomic_t *ref)
 
 #endif
 
+#define    SET_ATOMIC_FLAG(flag)  sys_atomic_set(flag, 1)
```
Comment (kgiusti) [inline include/qpid/dispatch/atomic.h:208]: Why not a CLEAR_ATOMIC_FLAG?

File: src/adaptors/http2/http2_adaptor.c [hunk 2]
```diff
@@ -2426,6 +2436,8 @@ static void handle_connection_event(pn_event_t *e, qd_server_t *qd_server, void
             send_settings_frame(conn);
             qd_log(log, QD_LOG_INFO, "[C%"PRIu64"] Accepted Ingress ((PN_RAW_CONNECTION_CONNECTED)) from %s", conn->conn_id, conn->remote_address);
         } else {
+            sys_atomic_init(&conn->raw_closed_read, 0);
```
Comment (kgiusti) [inline src/adaptors/http2/http2_adaptor.c:2439]: Why re-init?  Won't this destroy the existing state?

File: include/qpid/dispatch/atomic.h (hunk 1 above)
Comment (ganeshmurthy) [reply include/qpid/dispatch/atomic.h]: Introduced a CLEAR_ATOMIC_FLAG macro and the http adaptor now uses it to clear the flags once a new connection to the server is established.

File: src/adaptors/http2/http2_adaptor.c (hunk 2 above)
Comment (ganeshmurthy) [reply src/adaptors/http2/http2_adaptor.c]: Re-init was wrong. Now uses CLEAR_ATOMIC_FLAG

**Review by kgiusti — APPROVED** [review]

**Pull request closed without merging**